test/14_release-pinning.ts

```diff
@@ -546,5 +585,25 @@ describe(`Tracking latest release`, () => {
 				.delete({ resource: 'release', id: app5ReleaseId })
 				.expect(200);
 		});
+
+		it('should be able to delete an application where a device that is running its release has been moved to another application', async function () {
+			await pineUser
+				.patch({
+					resource: 'device',
+					id: device4.id,
+					body: {
+						is_running__release: app6ReleaseId,
+					},
+				})
+				.expect(200);
+			await pineUser.patch({
+				resource: 'device',
+				id: device4.id,
+				body: { belongs_to__application: application3Id },
+			});
+			await pineUser
+				.delete({ resource: 'application', id: applicationToDelete.id })
```

Comment:
`device4` was not in `applicationToDelete` though before.
How about provisioning a new one right here in the correct app?
```
			const device = await fakeDevice.provisionDevice(admin, applicationToDelete);
			await pineUser
				.patch({
					resource: 'device',
					id: device4.id,
					body: {
						is_running__release: app6ReleaseId,
					},
				})
				.expect(200);
			// ...
```